Assets/Scripts/Models/Time/TimeManager.cs

```diff
@@ -130,6 +134,16 @@ public void Update(float time)
         // Systems that update every frame.
         InvokeEvent(EveryFrame, time);
 
+        if (nextFrameActions.Count > 0)
+        {
+            for (int i = 0; i < nextFrameActions.Count; i++)
```

Comment:
We've been having a lot of issues with garbage collection, and apparently the version of mono used by Unity creates a *lot* of garbage with foreach, so if I can use a for statement, rather than a foreach, I will (and in fact have been changing other areas of code to use for in preference of foreach when possible)